### PR TITLE
Pm 1303 vpos step1 endpoint

### PR DESCRIPTION
**Pull request by DavideAmatoNexi**
#### List of Changes

Created new Endpoint `request-payments/vpos/{requestId}/method/notifications`
This API is used to return an html page during the "method" phase.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have created or updated an API interface.
  - [x] I have updated the OpenAPI spec
- [ ] I have created or updated a database table
  - [ ] I have updated both the Oracle and Postgres migrations
- [ ] My PR is cohesive and self-contained (rule of thumb: less than 250 changed lines) 

### Link to story

https://pagopa.atlassian.net/browse/PM-1303

### CODE REVIEW

File: src/main/java/it/pagopa/pm/gateway/utils/VPosRequestUtils.java [hunk 1]
```diff
@@ -36,6 +36,8 @@ public class VPosRequestUtils {
     private static final String METHOD = "/method";
     @Value("${vpos.request.responseUrl}")
     private String vposResponseUrl;
+    @Value("${pgs.vpos.method.notifyUrl}")
+    private String methodNotifyUrl;
 
     @Autowired
     VPosUtils vPosUtils;
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/pm/gateway/utils/VPosRequestUtils.java:43]: Prefer constructor injection to field injection :)

File: src/main/java/it/pagopa/pm/gateway/constant/VposConstant.java [hunk 2]
```diff
@@ -26,5 +26,13 @@ public class VposConstant {
...
-
+    public static final String HTML_METHOD = "<html>" +
+            "<head>" +
+            "<script>" +
+            "            var getUrl = window.location;" +
+            "window.parent.postMessage(\"3DS.Notification.Received\", getUrl.protocol + \"//\" + getUrl.host);" +
+            "</script>" +
+            "    </head>" +
+            "<body></body>" +
+            "</html>";
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/pm/gateway/constant/VposConstant.java:37]: Can we put this into a separate file?

File: api-specs/api_spec_PGS.yaml [hunk 3]
```diff
@@ -714,6 +714,24 @@ paths:
...
+      tags:
+        - Vpos
+      parameters:
+        - in: path
+          required: true
+          name: requestId
+          description: Id of the request
+          example: "41bc2409-5926-4aa9-afcc-797c7054e467"
+          schema:
+            type: string
```
[the remark this answers is not in the document]
Comment (DavideAmatoNexi) [reply api-specs/api_spec_PGS.yaml:729]: https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/202/commits/eb17612e54136d5bf1ba65baf4e89848a3581b44

File: src/main/java/it/pagopa/pm/gateway/utils/VPosRequestUtils.java (hunk 1 above)
Comment (DavideAmatoNexi) [reply src/main/java/it/pagopa/pm/gateway/utils/VPosRequestUtils.java]: https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/202/commits/573a1269a57687a7bad8255ca25c8dfa03a01aae

File: src/main/java/it/pagopa/pm/gateway/constant/VposConstant.java (hunk 2 above)
Comment (DavideAmatoNexi) [reply src/main/java/it/pagopa/pm/gateway/constant/VposConstant.java]: https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/202/commits/573a1269a57687a7bad8255ca25c8dfa03a01aae

File: src/main/resources/templates/methodNotification.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <script>
+		var getUrl = window.location;
+		window.parent.postMessage("3DS.Notification.Received", getUrl.protocol  "//"  getUrl.host);
```
Comment (giovanniberti) [inline src/main/resources/templates/methodNotification.html:6]: Missing `+`?
```suggestion
		window.parent.postMessage("3DS.Notification.Received", getUrl.protocol + "//"  + getUrl.host);
```

Also: [Location.origin](https://developer.mozilla.org/en-US/docs/Web/API/Location/origin) might be better suited here considered [`postMessage` API](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage)

Comment (DavideAmatoNexi) [reply src/main/resources/templates/methodNotification.html]: `location.origin` instead of `getUrl.protocol + "//" + getUrl.host`?

Comment (giovanniberti) [reply src/main/resources/templates/methodNotification.html]: Yess

File: api-specs/api_spec_PGS.yaml [hunk 5]
```diff
@@ -721,8 +721,8 @@ paths:
               $ref: '#/components/schemas/CreditCardResumeRequest'
         required: true
       responses:
-        "302":
-          description: "FOUND, Redirect to url"
+        "200":
+          description: "OK, executed asynchronous call to vpos"
```
Comment (giovanniberti) [inline api-specs/api_spec_PGS.yaml:725]: Is this change correct in the context of this PR?

File: src/main/resources/templates/methodNotification.html (hunk 4 above)
Comment (DavideAmatoNexi) [reply src/main/resources/templates/methodNotification.html]: https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/202/commits/24236697a28b209d3e4da0750547d620751699ce

**Review by giovanniberti — CHANGES_REQUESTED** [review]

File: api-specs/api_spec_PGS.yaml (hunk 5 above)
Comment (DavideAmatoNexi) [reply api-specs/api_spec_PGS.yaml:725]: Yes, it is correct

File: src/main/resources/templates/methodNotification.html [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <script>
+		var getUrl = window.location;
+		window.parent.postMessage("3DS.Notification.Received", location.origin);
```
Comment (giovanniberti) [inline src/main/resources/templates/methodNotification.html:6]: Either use the already declared variable `getUrl` (which, btw, could enjoy a better name maybe?) or use `window.location`

**Review by giovanniberti — CHANGES_REQUESTED** [review]

File: src/main/resources/templates/methodNotification.html (hunk 6 above)
Comment (DavideAmatoNexi) [reply src/main/resources/templates/methodNotification.html]: Something like that? https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/202/commits/0a171400565cb336c7e28250391917adffc2f4ab

File: src/main/resources/templates/methodNotification.html [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <script>
+		var getOrigin = window.location;
+		window.parent.postMessage("3DS.Notification.Received", getOrigin.origin);
```
Comment (giovanniberti) [inline src/main/resources/templates/methodNotification.html:6]: Idk, what about this?
```suggestion
		var getOrigin = window.location.origin;
		window.parent.postMessage("3DS.Notification.Received", getOrigin);
```

Comment (DavideAmatoNexi) [reply src/main/resources/templates/methodNotification.html:6]: yess, more clean

File: api-specs/api_spec_PGS.yaml (hunk 5 above)
Comment (DavideAmatoNexi) [reply api-specs/api_spec_PGS.yaml:725]: Yes, you are right i confused the PRs.
I can change it to "OK", such that the 2 api-specs line up

Comment (giovanniberti) [reply api-specs/api_spec_PGS.yaml:725]: Yes please, let's keep PRs self-consistent 😃

Comment (DavideAmatoNexi) [reply api-specs/api_spec_PGS.yaml]: i merged https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/200 into develop,  and i merged develop into this branch

**Pull request merged**